ark/analysis/spatial_analysis.py

```diff
@@ -116,6 +181,69 @@ def calculate_channel_spatial_enrichment(dist_matrices_dict, marker_thresholds,
     return values, stats
 
 
+def batch_cluster_spatial_enrichment(label_dir, all_data, batch_size=5, **kwargs):
```

Comment:
We could combine this function with previous batching function. We could add an "enrichment_type" argument, and if channels, have thresholds be a required argument. 

If you think that would be too messy can keep as is, but almost all the lines are identical

Reply:
So, it looks like there's a way to generate fov-batching versions of a bunch of functions (cell table generation, spatial enrichment, neighborhoods, etc) by using 'decorator-like' function factories.  It does look a little abstract, and is a little outside of the focus/scope of this PR, so I might leave these as separate for now, and submit a 'generic fov-batching' PR with that solution.